source/ed/tests/ed2nav_test.cpp

```diff
@@ -50,3 +53,26 @@ BOOST_AUTO_TEST_CASE(should_throw_on_bad_connection_string) {
     const char* argv[] = {"ed2nav_test", "--connection-string=\"blahblah\""};
     BOOST_CHECK_THROW(ed::ed2nav(2, argv), std::exception);
 }
+
+BOOST_AUTO_TEST_CASE(writing_test_ed2nav_should_not_throw) {
```

Comment:
Can you add failing tests?
(like rename impossible, file not generated, etc.)

Reply:
Added a test on file save

Reply:
Unfortunately, making a call to `rename` to fail (in a decent way), would mean to mock the free function which is very complicated without a proper mocking  framework :cry: